### DISCUSSION TITLE
Input.gff does not exist. ?

### DISCUSSION THREAD

**Discussion by bveyb** [NBISweden/AGAT#326] [Q&A] [answered]
Hi,
I'd try to use the docker image for a conversion of the actual TAIR10 Arabidopsis GFF3 annotation file to GTF,
but the agat_convert_sp_gff2gtf.pl does not find/open my GFF file:

> docker run --rm -v $(pwd):/app quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0 agat_convert_sp_gff2gtf.pl --gff   
   $(pwd)/GCF_000001735.4_TAIR10.1_genomic.gff -o GCF_000001735.4_TAIR10.1_genomic.gtf

Using standard /usr/local/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file
Update config
Reading input file
********************************************************************************
*                              - Start parsing -                               *
********************************************************************************
-------------------------- parse options and metadata --------------------------
=> Accessing the feature_levels YAML file
Using standard /usr/local/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file
=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):
	* locus_tag
	* gene_id
=> merge_loci option deactivated
=> Machine information:
	This script is being run by perl v5.32.1
	Bioperl location being used: /usr/local/lib/perl5/site_perl/Bio/
	Operating system being used: linux 
=> Accessing Ontology
	No ontology accessible from the gff file header!
	We use the SOFA ontology distributed with AGAT:
		/usr/local/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo
	Read ontology /usr/local/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:
		4 root terms, and 2596 total terms, and 1516 leaf terms
	Filtering ontology:
		We found 1861 terms that are sequence_feature or is_a child of it.
--------------------------------- parsing file ---------------------------------
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!/home/bveyb/WT_28/GCF_000001735.4_TAIR10.1_genomic.gff does not exist. Please !
!                       verify the input file name/path                        !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

What should I do?

**Comment by Juke34** [comment, marked as answer]
Thank you for using AGAT.
Using containers work slightly differently than a simple script. When using docker you must load/mount the data within the container, otherwise it is not accessible.

You can do something like that:

```
docker run -v ${PWD}:/agat quay.io/biocontainers/agat:0.8.0--pl5262hdfd78af_0 agat_convert_sp_gxf2gxf.pl --gff /agat/path/to/my/folder/test.gff -o /agat/test.gtf
```

**Reply by bveyb** [reply]
Thank you much for the fast reply.

I'd try

```
docker run -v ${PWD}:/agat quay.io/biocontainers/agat:0.8.0--pl5262hdfd78af_0 agat_convert_sp_gxf2gxf.pl --gff /home/bveyb/WT_28/GCF_000001735.4_TAIR10.1_genomic.gff -o GCF_000001735.4_TAIR10.1_genomic.gtf
```
-------------------------------------------------------------------------------
This script is being run by perl v5.26.2
Bioperl location being used: /usr/local/lib/site_perl/5.26.2/Bio/
-------------------------------------------------------------------------------
********************************************************************************
*                              - Start parsing -                               *
********************************************************************************
-------------------------- parse options and metadata --------------------------
=> Accessing the feature level json files
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level1.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level2.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level3.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_spread.json file
=> Attribute used to group features when no Parent/ID relationship exists:
	* locus_tag
	* gene_id
=> merge_loci option deactivated
=> Accessing Ontology
	No ontology accessible from the gff file header!
	We use the SOFA ontology distributed with AGAT:
		/usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo
	Read ontology /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo:
		4 root terms, and 2472 total terms, and 1436 leaf terms
	Filtering ontology:
		We found 1757 terms that are sequence_feature or is_a child of it.
-------------------------------- parse features --------------------------------
Info: Cannot count total line number with builtin wc. Consequently progress bar unavailable.
cannot open file /home/bveyb/WT_28/GCF_000001735.4_TAIR10.1_genomic.gffDied at /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm line 3243.

Didn't work.
So, I'd tried
```
docker run -v ${PWD}:/agat quay.io/biocontainers/agat:0.8.0--pl5262hdfd78af_0 agat_convert_sp_gff2gtf.pl --gff /home/bveyb/WT_28/GCF_000001735.4_TAIR10.1_genomic.gff -o GCF_000001735.4_TAIR10.1_genomic.gtf
```

converting to GTF3
********************************************************************************
*                              - Start parsing -                               *
********************************************************************************
-------------------------- parse options and metadata --------------------------
=> Accessing the feature level json files
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level1.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level2.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level3.json file
	Using standard /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_spread.json file
=> Attribute used to group features when no Parent/ID relationship exists:
	* locus_tag
	* gene_id
=> merge_loci option deactivated
=> Accessing Ontology
	No ontology accessible from the gff file header!
	We use the SOFA ontology distributed with AGAT:
		/usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo
	Read ontology /usr/local/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo:
		4 root terms, and 2472 total terms, and 1436 leaf terms
	Filtering ontology:
		We found 1757 terms that are sequence_feature or is_a child of it.
-------------------------------- parse features --------------------------------
Info: Cannot count total line number with builtin wc. Consequently progress bar unavailable.
readline() on closed filehandle $fh at /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm line 3245.
readline() on closed filehandle $fh at /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm line 3275.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!                                   Doesn't look like a GFF file                                   !
!             Let's see what the Bioperl parser can do with that...(using gff3 parser)             !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=> GFF version parser used: 3

------------- EXCEPTION: Bio::Root::Exception -------------
MSG: Could not read file '/home/bveyb/WT_28/GCF_000001735.4_TAIR10.1_genomic.gff': No such file or directory
STACK: Error::throw
STACK: Bio::Root::Root::throw /usr/local/lib/site_perl/5.26.2/Bio/Root/Root.pm:447
STACK: Bio::Root::IO::_initialize_io /usr/local/lib/site_perl/5.26.2/Bio/Root/IO.pm:268
STACK: Bio::Tools::GFF::new /usr/local/lib/site_perl/5.26.2/Bio/Tools/GFF.pm:200
STACK: AGAT::OmniscientI::slurp_gff3_file_JD /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm:306
STACK: /usr/local/bin/agat_convert_sp_gff2gtf.pl:87

I tried the newest version (quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0) too.
Still the same problem, it is not able to read the file.

Due you have any clue?
Thank you.

**Comment by Juke34** [comment]
Try

`docker run -v ${PWD}:/agat quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0 agat_convert_sp_gxf2gxf.pl --gff /agat//test.gff -o /agat/test.gtf`

**Reply by bveyb** [reply]
Thank you.

Some Exons (34) and a lot of UTRs (104734) were created,
but it seems, conversion was successful:
=> OmniscientI total time: 128 seconds
GFF3 file parsed
Formating output to GFF3
usage: /usr/local/bin/agat_convert_sp_gxf2gxf.pl --gff /agat//GCF_000001735.4_TAIR10.1_genomic.gff -o /agat/GCF_000001735.4_TAIR10.1_genomic.gtf
Job done in 170 seconds.

Have a good weekend.
